src/App/services/Collisions/Collisions.ts

```diff
@@ -78,88 +78,57 @@ export class Collisions implements IService {
-         // }
-         const attrValue = this.attributes.getAttribute({
-            gameObjectId: enemy.id,
-            attribute: "collisionType"
+      const enemiesThatCanCollide = Object.values(this.enemies.enemies)
+         .filter((enemy) => {
+            const collisionType = this.attributes.getAttribute({
+               gameObjectId: enemy.id,
+               attribute: "collisionType"
+            });
+            return collisionType !== "none";
          });
-         switch(attrValue){
-            case "enemy":
-               enemies.push(enemy);
-               break;
-            case "enemyBullet":
-               enemyBullets.push(enemy);
-               break;
-            case "playerBullet":
-               playerBullets.push(enemy);
-               break;
-            default:
-               // NOOP
-         }
-      }
-
-      const player = this.enemies.player;
 
-      // Observe: by enemy and not by enemy bullets.
-      const playerWasHitByEnemy =
-         this.calcCollisions({
-            doesThis: player,
-            collideWithThese: enemies
-         }).collided;
-
-      if(playerWasHitByEnemy) {
-         allGameObjectsThatWereHit.push(player.id);
-      }
-      
-      const enemiesHitByPlayerBullets: string[] = [];
-      for(const enemy of enemies) {
-         const collision = this.calcCollisions({
-            doesThis: enemy,
-            collideWithThese: playerBullets
-         });
-         // adds both the "enemy" and what it collides with (for example a playerBullet).
-         if(collision.collided) {
-            enemiesHitByPlayerBullets.push(enemy.id, collision.collidedWithId);
-         }
-      }
+      for (const enemy1 of enemiesThatCanCollide) {
+         // const attrValue = this.attributes.getAttribute({
+         //    gameObjectId: enemy.id,
+         //    attribute: "collisionType"
+         // });
+         
+         collisions[enemy1.id] = [];
+
+         for( const enemy2 of enemiesThatCanCollide) {
+            if(enemy1.id === enemy2.id) {
+               continue; // dont check collision with self.
+            }
 
-      const enemyBulletsThatHitPlayer: string[] = [];
-      for(const enemyBullet of enemyBullets) {
-         const collision = this.calcCollisions({
-            doesThis: enemyBullet,
-            collideWithThese: [player]
-         });
-         if(collision.collided) {
-            enemyBulletsThatHitPlayer.push(enemyBullet.id);
+            const collided = this.calcCollision({
+               doesThis: enemy1,
+               collideWithThis: enemy2
+            });
+
+            if(collided) {
+               const collisionType = assertString(
+                  this.attributes.getAttribute({
+                     gameObjectId: enemy2.id,
+                     attribute: "collisionType"
+                  })
+               );
+               collisions[enemy1.id] = [...new Set(
+                  [...collisions[enemy1.id], collisionType]
```

Comment:
seems I only record one of each collision type. it's not obvious that that's what I want. What about the player collidimg with 2 powerups and 1 enemy?

Reply:
explain in comment